lib/test/t_Trapezoidal_std.cxx

```diff
@@ -43,140 +43,159 @@ int main(int, char *[])
     // Test basic functionnalities
     checkClassWithClassName<TestObject>();
 
+    Collection< Trapezoidal > coll;
+    coll.add(Trapezoidal(1.0, 1.2, 3.0, 14.0));   // a<>b<>c<>d
+    coll.add(Trapezoidal(1.0, 1.0, 3.0, 14.0));   // a==b<>c<>d
+    coll.add(Trapezoidal(1.0, 1.2, 1.2, 14.0));   // a<>b==c<>d
+    coll.add(Trapezoidal(1.0, 1.0, 1.0, 14.0));   // a==b==c<>d
+    coll.add(Trapezoidal(1.0, 1.2, 14.0, 14.0));  // a<>b<>c==d
+    coll.add(Trapezoidal(1.0, 1.0, 14.0, 14.0));  // a==b<>c==d
+    coll.add(Trapezoidal(1.0, 14.0, 14.0, 14.0)); // a<>b==c==d
+
     // Instantiate one distribution object
-    Trapezoidal distribution(1.0, 1.2, 3.0, 14.0);
-    fullprint << "Distribution " << distribution << std::endl;
-    std::cout << "Distribution " << distribution << std::endl;
-
-    // Is this distribution elliptical ?
-    fullprint << "Elliptical = " << (distribution.isElliptical() ? "true" : "false") << std::endl;
-
-    // Is this distribution continuous ?
-    fullprint << "Continuous = " << (distribution.isContinuous() ? "true" : "false") << std::endl;
-
-    // Test for realization of distribution
-    Point oneRealization = distribution.getRealization();
-    fullprint << "oneRealization=" << oneRealization << std::endl;
-
-    // Test for sampling
-    UnsignedInteger size = 10000;
-    Sample oneSample = distribution.getSample( size );
-    fullprint << "oneSample first=" << oneSample[0] << " last=" << oneSample[size - 1] << std::endl;
-    fullprint << "mean=" << oneSample.computeMean() << std::endl;
-    fullprint << "covariance=" << oneSample.computeCovariance() << std::endl;
-    fullprint << "skewness=" << oneSample.computeSkewness() << std::endl;
-    fullprint << "kurtosis=" << oneSample.computeKurtosis() << std::endl;
-    size = 100;
-    for (UnsignedInteger i = 0; i < 2; ++i)
+    for (UnsignedInteger nTrapezoidal = 0; nTrapezoidal < coll.getSize(); ++nTrapezoidal)
     {
-      fullprint << "Kolmogorov test for the generator, sample size=" << size << " is " << (FittingTest::Kolmogorov(distribution.getSample(size), distribution).getBinaryQualityMeasure() ? "accepted" : "rejected") << std::endl;
-      size *= 10;
+      // Instantiate one distribution object
+      Trapezoidal distribution(coll[nTrapezoidal]);
+      fullprint << "Distribution " << distribution << std::endl;
+      std::cout << "Distribution " << distribution << std::endl;
+
+      // Is this distribution elliptical ?
+      fullprint << "Elliptical = " << (distribution.isElliptical() ? "true" : "false") << std::endl;
```

Comment:
This sort of check could be made using assertions.

Reply:
As discussed during our last technical meeting, you are right, it could be done using assertions, but as there are two to four such tests in each of the 80 tests related to distributions, I don't see the need to cherry-pick some parts of some tests to convert them to assertions, and unfortunately I don't have the time to convert this test to the new format as a whole...